proposals/extensions.md

```diff
@@ -1158,6 +1158,205 @@ It allows roundtripping of extension symbols through metadata (full and referenc
 For example: `implicit extension R for UnderlyingType` yields
 `public static void <ImplicitExtension>$(UnderlyingType)`.  
 
+Instance method/property/indexer declarations in source are represented as static declarations in metadata:
```

Comment:
Let's add open issues at the top for the follow-up questions (how to expose the new APIs to other compilers, capturing receiver in type parameter scenario, ...).